facebook/facebook-for-woocommerce
# Adds HPOS compatibility

### Changes proposed in this Pull Request:



Closes #2260.

Thank you @mikkamp for your research on HPOS compatibility. This PR is built on top of @mikkamp's inputs. Since the HPOS upgrade recipe is updated, I ran the check again. I concur with the findings in #2260.

This PR:

* Adds a check to set the `screen_id` based on order data store. The `screen_id` is used to enqueue `facebook-for-woocommerce-modal`, `wc-facebook-google-product-category-fields` and to render the `render_modal_template`.
* Sets the order and orders page URL in the acceptance test.
* Sets the HPOS compatibility to `true`.


### Detailed test instructions:


1. Enable HPOS and initial sync of orders by following the instructions given here: https://github.com/woocommerce/woocommerce/wiki/High-Performance-Order-Storage-Upgrade-Recipe-Book#how-to-enable-hpos
2. Generate the build for this branch and install.
3. Navigate to WooCommerce > Settings > Advanced > Custom data stores and set **Data store for orders** option to `Use the WooCommerce orders tables`.
4. Navigate to WooCommerce > Orders and click on any single order. 
5. Click on view source and verify `facebook-for-woocommerce-modal`, `wc-facebook-google-product-category-fields` and `tmpl-facebook-for-woocommerce-modal` are enqueued.
6. Navigate to WooCommerce > Settings > Advanced > Custom data stores and set **Data store for orders** option to `Use the WordPress posts table`.
7. Navigate to WooCommerce > Orders and click on any single order. 
8. Click on view source and verify `facebook-for-woocommerce-modal`, `wc-facebook-google-product-category-fields` and `tmpl-facebook-for-woocommerce-modal` are enqueued.

### Additional details:

As noted in #2260, there are changes that need to be addressed in the SkyVerge framework for HPOS compatibility. However, based on internal discussions, the part of the framework code that requires changing is not used by the plugin. I need additional eyes here to verify that the matches listed in this comment https://github.com/woocommerce/facebook-for-woocommerce/issues/2260#issuecomment-1295105024 do not affect the plugin.

I am unable to run the Acceptance test which has 2 other code changes.


### Changelog entry

> Add - HPOS Compatibility
